src/material-experimental/mdc-slider/slider.ts

```diff
@@ -201,29 +330,28 @@ export class MatSlider implements AfterViewInit, OnDestroy {
 
   /** Gets the slider thumb HTML element of the given thumb. */
   _getThumbElement(thumb: Thumb): HTMLElement {
-    const thumbs = this._thumbs.toArray().map(e => e.nativeElement);
-    return thumb === Thumb.END ? thumbs[thumbs.length - 1] : thumbs[0];
+    const thumbElementRef = thumb === Thumb.END ? this._thumbs.last : this._thumbs.first;
+    return thumbElementRef.nativeElement;
   }
 
   /** Gets the slider knob HTML element of the given thumb. */
   _getKnobElement(thumb: Thumb): HTMLElement {
-    const knobs = this._knobs.toArray().map(e => e.nativeElement);
-    return thumb === Thumb.END ? knobs[knobs.length - 1] : knobs[0];
+    const knobElementRef = thumb === Thumb.END ? this._knobs.last : this._knobs.first;
+    return knobElementRef.nativeElement;
   }
 
   /**
-   * Gets the text representation of the given value.
+   * Sets the value indicator text of the given thumb using the given value.
    *
-   * Uses the `displayWith` function if one has been provided. Otherwise, it just returns the
-   * current numeric value as a string.
+   * Uses the `displayWith` function if one has been provided. Otherwise, it just uses the
+   * numeric value as a string.
    */
-  _getValueIndicatorText(value: number): string {
-    return this.displayWith ? this.displayWith(value) : value.toString();
-  }
-
-  /** Gets the text representation of the current value of the given thumb. */
-  _getValueIndicatorTextByThumb(thumb: Thumb): string {
-    return this._getValueIndicatorText(this._getValue(thumb));
+  _setValueIndicatorText(value: number, thumb: Thumb): void {
+    const valueIndicatorTextElementRef = thumb === Thumb.END
+        ? this._valueIndicatorTextElements.last
+        : this._valueIndicatorTextElements.first;
+    const valueText = this.displayWith ? this.displayWith(value) : `${value}`;
+    valueIndicatorTextElementRef.nativeElement.textContent = valueText;
```

Comment:
Why change this from a text binding to directly changing the DOM? If it's strictly necessary, we should add a comment here explaining why